packages/headless/src/features/commerce/common/base-commerce-api-request-builder.ts

```diff
@@ -32,21 +35,15 @@ export const buildBaseCommerceAPIRequest = (
     organizationId: state.configuration.organizationId,
     trackingId: state.configuration.analytics.trackingId!,
     ...restOfContext,
-    ...(state.configuration.analytics.enabled
-      ? {clientId: navigatorContext.clientId}
-      : {}),
+    ...(state.configuration.analytics.enabled ? {clientId} : {}),
     context: {
       user: {
         ...location,
-        ...(navigatorContext.userAgent
-          ? {userAgent: navigatorContext.userAgent}
-          : {}),
+        ...(userAgent ? {userAgent} : {}),
       },
       view: {
         ...view,
-        ...(navigatorContext.referrer
-          ? {referrer: navigatorContext.referrer}
-          : {}),
+        ...(referrer ? {referrer} : {}),
```

Comment:
Making the change at this level means relay will always be responsible for populating the referrer, and any custom navigator context provided by the user will be ignored. Is this ok? I think the ability to customize was needed for SSR scenario?